### ISSUE TITLE
Time zone rejected in timestamps

### ISSUE THREAD

**Issue by ningit** [systemd/systemd#34670]
### systemd version the issue has been seen with

256.6-1

### Used distribution

ArchLinux

### Linux kernel version used

6.11.2-arch1-1

### CPU architectures issue was seen on

x86_64

### Component

systemctl, systemd-analyze

### Expected behaviour you didn't see

The manual page for timestamps (section *parsing timestamps*) cites `Fri 2012-11-23 23:02:15 CET` as an example of valid timestamp in the first paragraph, and tells in the second one that time zones "*may be given after a space [...] it can be: "UTC", an entry in the installed IANA timezone database ("CET", [...])*".

The end of the section tells timestamps can be validated with `systemd-analyze timestamp`. Hence, the reader would expect that the following command validates the timestamp string.
```
$ systemd-analyze timestamp 'Fri 2012-11-23 23:02:15 CET'
  Original form: Fri 2012-11-23 23:02:15 CET
Normalized form: Fri 2012-11-23 23:02:15 CET
       (in UTC): Fri 2012-11-23 22:02:15 UTC
   UNIX seconds: @1353708135
       From now: 11 years 10 months ago
```

### Unexpected behaviour you saw

However, the command produces this output:
```
$ systemd-analyze timestamp 'Fri 2012-11-23 23:02:15 CET'
Failed to parse "Fri 2012-11-23 23:02:15 CET": Invalid argument
Hint: this expression is a valid calendar specification. Use 'systemd-analyze calendar "Fri 2012-11-23 23:02:15 CET"' instead
```

This timestamp is also rejected in other `systemd` commands. For example,
```
$ systemctl poweroff --when='Fri 2012-11-23 23:02:15 CET'
Failed to parse --when= argument 'Fri 2012-11-23 23:02:15 CET': Invalid argument
```

### Steps to reproduce the problem

Run the commands above.

**Comment by arvidjaar** [comment]
Same error on Ubuntu 24.04 with systemd 255.4.

**Comment by yuwata** [comment]
Ouch. Fix is waiting in #34672.
